h2o-admissibleml/src/main/java/hex/Infogram/Infogram.java

```diff
@@ -0,0 +1,408 @@
+      }
+      if (_parms._total_information_threshold != -1) {
+        warn("total_information_threshold", "Should not set total_information_threshold for fair" +
+                " infogram runs, set relevance_index_threshold instead.  Using default of 0.1 if not set");
+      }
+    }
+
+    // check top k to be between 0 and training dataset column number
+    if (_parms._top_n_features < 0)
+      error("_topk", "topk must be between 0 and the number of predictor columns in your training dataset.");
+
+    _numPredictors = _parms.train().numCols()-1;
+    if (_parms._weights_column != null)
+      _numPredictors--;
+    if (_parms._offset_column != null)
+      _numPredictors--;
+    if ( _parms._top_n_features > _numPredictors) {
+      warn("_topk", "topk exceed the actual number of predictor columns in your training dataset." +
+              "  It will be set to the number of predictors in your training dataset.");
+      _parms._top_n_features = _numPredictors;
+    }
+
+    if (_parms._nparallelism < 0)
+      error("nparallelism", "must be >= 0.  If 0, it is adaptive");
+
+    if (_parms._nparallelism == 0) // adaptively set nparallelism
+      _parms._nparallelism = H2O.NUMCPUS;
+    
+    if (_parms._compute_p_values)
+      error("compute_p_values", " compute_p_values calculation is not yet implemented.");
+    
+    if (nclasses() < 2)
+      error("distribution", " infogram currently only supports classification models");
+    
+    if (DistributionFamily.AUTO.equals(_parms._distribution)) {
+      _parms._distribution = (nclasses() == 2) ? DistributionFamily.bernoulli : DistributionFamily.multinomial;
+    }
+  }
+
+  private class InfogramDriver extends Driver {
+    void prepareModelTrainingFrame() {
+      _generatedFrameKeys = new ArrayList<>(); // generated infogram model plus one for safe Infogram
+      String[] eligiblePredictors = extractPredictors(_parms);  // exclude senstive attributes if applicable
+      _baseOrSensitiveFrame = extractTrainingFrame(_parms, _parms._protected_columns, 1, _parms.train().clone());
+      _parms.extraModelSpecificParams(); // copy over model specific parameters to build infogram
+      _topKPredictors = extractTopKPredictors(_parms, _parms.train(), eligiblePredictors, _generatedFrameKeys); // extract topK predictors
+      _numModels = 1 + _topKPredictors.length;
+      _modelDescription = generateModelDescription(_topKPredictors, _parms._protected_columns);
+    }
+
+    @Override
+    public void computeImpl() {
+      init(true);
+      if (error_count() > 0)
+        throw H2OModelBuilderIllegalArgumentException.makeFromBuilder(Infogram.this);
+      _job.update(0, "Initializing model training");
+      prepareModelTrainingFrame(); // generate training frame with predictors and sensitive features (if specified)
+      buildModel();
+    }
+    // todo:  add max_runtime_secs restrictions
```

Comment:
is this todo to be done in this PR or later?

Reply:
Not here, later in here: https://h2oai.atlassian.net/browse/PUBDEV-8464